src/gui/sdlmain.cpp

```diff
@@ -3832,11 +3833,30 @@ bool GFX_Events()
 				assert(sdl.desktop.window.width > 0 &&
 				       sdl.desktop.window.height > 0);
 
-				log_display_properties(sdl.draw.render_width_px,
-				                       sdl.draw.render_height_px,
-				                       sdl.video_mode,
-				                       sdl.rendering_backend);
-				continue;
+				// SDL_WINDOWEVENT_RESIZED events are sent twice
+				// on macOS when resizing the window, so we're
+				// "debouncing" the duplicate logging with a timer.
+				using namespace std::chrono;
+				using clock = steady_clock;
+
+				static clock::time_point last_resized = {};
+
+				const auto now = clock::now();
+				const auto elapsed_ms = duration_cast<milliseconds>(
+				                                now - last_resized)
+				                                .count();
+
+				const auto log_debounce_threshold_ms = 10;
+
+				if (elapsed_ms > log_debounce_threshold_ms) {
+					log_display_properties(sdl.draw.render_width_px,
+					                       sdl.draw.render_height_px,
+					                       sdl.video_mode,
+					                       sdl.rendering_backend);
+				}
+
+				last_resized = clock::now();
+			}	continue;
```

Comment:
If there are duplicate resize events, can we get to the bottom of why that's happening?

Worst case: compare the existing size settings to the incoming request, and skip if it's asking for the same settings.

Reply:
Not rapid resize. Just a single normal resize.

`SDL_WINDOWEVENT_RESIZED` is supposed to be fired _once_ once you've stopped resizing. Note `RESIZED` (paste tense) in the name.

...but on macOS (at least) it fires twice. It's either an SDL bug, or a macOS thing, or any number of other SDL/OS interactions, but it's firmly outside of our control.

Reply:
We can't compare the incoming settings because at this point we've already adjusted all the `sdl.*` structs to the new settings.

Reply:
So, I'm not getting into the whole complicated SDL maze now; I'd much rather put that effort into rewriting this from scratch. So, two options:

1) Leave the timer hack in; it does the job, and the user observable behaviour is what matter.
2) Remove the time hack, then we'll get the display settings logged twice when the window is resized 🤷🏻 

I'm fine with either 😄

Reply:
I don't like either 😅
I'd rather dedupe logically than make it time based. 
The logging function can hold static `last_`* args that are compared against the incoming args.
That way any duplicate logging will never happen, and likewise all unique sizes will always be logged.

Reply:
Right, now I understood what you asked for 😄 Okay, I've rewritten it like that and it's still getting rid of the dupes.